UICatalog/Scenarios/Scrolling.cs

```diff
@@ -155,9 +155,9 @@ public override void Setup ()
 			void Top_Loaded (object sender, EventArgs args)
 			{
 				// BUGBUG: v2 - this broke somehow - fix later
-				//horizontalRuler.Text = rule.Repeat ((int)Math.Ceiling ((double)(horizontalRuler.Bounds.Width) / (double)rule.Length)) [0..(horizontalRuler.Bounds.Width)] +
-				//	"\n" + "|         ".Repeat ((int)Math.Ceiling ((double)(horizontalRuler.Bounds.Width) / (double)rule.Length)) [0..(horizontalRuler.Bounds.Width)];
-				//verticalRuler.Text = vrule.Repeat ((int)Math.Ceiling ((double)(verticalRuler.Bounds.Height * 2) / (double)rule.Length)) [0..(verticalRuler.Bounds.Height * 2)];
+				horizontalRuler.Text = rule.Repeat ((int)Math.Ceiling ((double)(horizontalRuler.Bounds.Width) / (double)rule.Length)) [0..(horizontalRuler.Bounds.Width)] +
```

Comment:
Let's use the new `Ruler` class here.

Reply:
@tig I don't understand how you want I draw the label inside the scroll view here, instead of fill the label text once. Did you make same confusion or I'm missing something?

Reply:
I just hate duplicate code. I wrote `Ruler` recently so that it could be used anywhere we want a ruler.

Reply:
I now understand what you mean. I created the `StringExtensions` as public to be available where is needed.